src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,55 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installURL, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installURL, this.statusChanged);
+    this.addonManager
+      .getAddon()
```

Comment:
I think that looks weird. I like to keep things horizontal or vertical, not a mix of both. If we're putting `.getAddon()` on the line above I don't see how `.then(() => {` makes it any worse.